### ISSUE TITLE
unit goes to `BlockedStatus` on conditions that cannot be resolved by a human operator

### ISSUE THREAD

**Issue by DnPlas** [canonical/kubeflow-profiles-operator#101]
This charm goes into `BlockedStatus` when:

* The charm is attempting to [update layers](https://github.com/canonical/kubeflow-profiles-operator/blob/main/src/charm.py#L218)
* The workload container is [not ready](https://github.com/canonical/kubeflow-profiles-operator/blob/main/src/charm.py#L223) (see also [here](https://github.com/canonical/kubeflow-profiles-operator/blob/main/src/charm.py#L240))

A `BlockedStatus` means a human operator is needed for unblocking the unit through certain actions, in the case of the above, there is little to nothing a human operator can do, causing the unit to be in such state without a proper resolution. 

#### Possible fix

Change the status to `Waiting` or `Error` when needed with appropriate messaging.

**Comment by DnPlas** [comment]
This is a duplicate of canonical/bundle-kubeflow#549
